packages/core/src/command/rate-limiting/ddb-store.ts

```diff
@@ -0,0 +1,323 @@
+      partition: this.prefixCountKey(key),
+      expression: "ADD totalHits :inc SET resetTime = if_not_exists(resetTime, :resetTime)",
+      expressionAttributesValues: {
+        ":inc": 1,
+        ":resetTime": buildDayjs().add(this.windowMs, "milliseconds").unix(),
+      },
+    });
+    const entry = rateLimitCountSchema.safeParse(item.Attributes);
+    if (!entry.success) {
+      const error = entry.error;
+      const msg = "Error parsing DynamoStore count entry";
+      log(`${msg} - error: ${errorToString(error)}`);
+      capture.error(msg, {
+        extra: {
+          key,
+          context: "dynamo-store.increment",
+          error,
+        },
+      });
+      throw new MetriportError(msg, error, { method: "increment", key });
+    }
+    if (!entry.data.resetTime) {
+      throw new MetriportError("resetTime not set correctly", undefined, {
+        method: "increment",
+        key,
+      });
+    }
+    return {
+      totalHits: entry.data.totalHits,
+      resetTime: this.buildDate(entry.data.resetTime),
+    };
+  }
+
+  /**
+   * Method to decrement a client's hit counter.
+   *
+   * @param key {string} - The identifier for a client.
+   *
+   * @public
+   */
+  async decrement(key: string): Promise<void> {
+    const { log } = out(`DynamoStore decrement - key ${key}`);
+    const item = await this.ddbUtils.update({
+      partition: this.prefixCountKey(key),
+      expression: "ADD totalHits :inc",
+      expressionAttributesValues: {
+        ":inc": -1,
+      },
+    });
+    const entry = rateLimitCountSchema.safeParse(item.Attributes);
+    if (!entry.success) {
+      const error = entry.error;
+      const msg = "Error parsing DynamoStore count entry";
+      log(`${msg} - error: ${errorToString(error)}`);
+      capture.error(msg, {
+        extra: {
+          key,
+          context: "dynamo-store.decrement",
+          error,
+        },
+      });
+      throw new MetriportError(msg, error, { method: "decrement", key });
+    }
+  }
+
+  /**
+   * Method to reset a client's hit counter.
+   *
+   * @param key {string} - The identifier for a client.
+   *
+   * @public
+   */
+  async resetKey(key: string): Promise<void> {
+    const { log } = out(`DynamoStore resetKey - key ${key}`);
+    const item = await this.ddbUtils.update({
+      partition: this.prefixCountKey(key),
+      expression: "SET totalHits = :totalHits REMOVE resetTime",
+      expressionAttributesValues: {
+        ":totalHits": 0,
+      },
+    });
+    const entry = rateLimitCountSchema.safeParse(item.Attributes);
+    if (!entry.success) {
+      const error = entry.error;
+      const msg = "Error parsing DynamoStore count entry";
+      log(`${msg} - error: ${errorToString(error)}`);
+      capture.error(msg, {
+        extra: {
+          key,
+          context: "dynamo-store.resetKey",
+          error,
+        },
+      });
+      throw new MetriportError(msg, undefined, { method: "resetKey", key });
+    }
```

Comment:
ditto: nothing done w/ the result of parsing

Reply:
removed.